mobx/test/observable_list_test.dart

```diff
@@ -14,37 +17,59 @@ void main() {
 
       expect(count, equals(0));
 
-      list.add(observable(20));
+      list.add(20);
       expect(count, equals(1));
       d();
     });
 
+    test('cast returns a live view to list', () {
+      final list = ObservableList<num>.of(<num>[0, 1, 2, 3, 4, 5, 6]);
+      final casted = list.cast<int>();
+
+      var count = 0;
+      autorun((_) {
+        // ignore:unnecessary_statements
+        casted[1];
+        count++;
+      });
+      expect(count, equals(1));
+
+      list[1] = 99;
+      expect(casted[1], equals(99));
+      expect(count, equals(2));
+    });
+
+    test('Autorun should execute when items are added to an empty list', () {
+      final list = ObservableList<int>();
+
+      var count = 0;
+      autorun((_) {
+        for (final v in list) {
+          count++;
+        }
+      });
+      expect(count, equals(0));
+
+      list.add(0);
+      expect(count, equals(1));
+    });
+
     group('fires reportObserved() for read-methods', () {
       <String, Function(ObservableList<int>)>{
-        'map': (_) => _.map((x) => x),
         'isEmpty': (_) => _.isEmpty,
         'isNotEmpty': (_) => _.isNotEmpty,
         'iterator': (_) => _.iterator,
-        'reversed': (_) => _.reversed,
         'single': (_) => _ignoreException(() => _.single),
         'first': (_) => _ignoreException(() => _.first),
         'last': (_) => _ignoreException(() => _.last),
         'toSet': (_) => _.toSet(),
         'toList': (_) => _.toList(),
```

Comment:
Seem to have skipped a few methods like `reversed`, `cast`, etc. Intentional ?

Reply:
Their reports are now lazy, so calling `observableList.map(fn)` doesn't report yet, but iterating the iterable returned by `map` will, e.g. ` observableList.map(fn).forEach((_) {})`.

The `fires reportObserved() lazily on iterator returning methods` test group handles those.